### PR TITLE
Rewrite in ES6, add typescript definitions

### PR DESCRIPTION
**Pull request by LKay**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { createTransform } from "redux-persist";
+import LZ from "lz-string";
+import stringify from "json-stringify-safe";
+import process from "process";
```
Comment (rt2zz) [inline src/index.js]: I think this is unnecessary. Afaict `process` module is intended to provide nextTick functionality, process.env already exists.

**Comment by rt2zz** [comment]
awesome! one little comment about the process module, otherwise looks great.

File: src/index.js (hunk 1 above)
Comment (LKay) [inline src/index.js]: I'm not so sure about that in browser environment. I had some problem with `process` (and also `env`) being undefined in some browsers.

Comment (rt2zz) [inline src/index.js]: hm, yes that is true, I suppose the best solution here would be to do it like redux https://github.com/reactjs/redux/blob/master/src/combineReducers.js#L5

redux-persist I think needs the same update

Comment (LKay) [inline src/index.js]: Looks legit, I'll change that.

**Comment by LKay** [comment]
I made necessary changes. Also updated PR in the main project [#281](https://github.com/rt2zz/redux-persist/pull/281).

**Comment by rt2zz** [comment]
🎈awesome!

**Pull request merged**